### DISCUSSION TITLE
Where is the system name gathered from

### DISCUSSION THREAD

**Discussion by redat00** [lldpd/lldpd#549] [Q&A] [answered]
Hi everyone ! 

I have a machine which is running `lldpd` and it works great ! I can see the machine on the switch it is connected to, but the only problem is that it is returning `localhost` as the system name, instead of the actual hostname of the machine.. 

I checked everywhere (/etc/hosts, /etc/hostname...) and nowhere it is set to `localhost`.. 
So my question is : Where does lldpd find the system name information ?

**Comment by vincentbernat** [comment, marked as answer]
It does `getent hosts $(uname -n)`. So, it's a combination of `/etc/hostname` (result of `uname -n`) and `/etc/hosts`.
